src/journal/journal.c

```diff
@@ -88,9 +91,11 @@ static void strip_new_line(char *l)
 {
         size_t len;
 
-        len = strlen(l);
-        if (*(l+len-1) == '\n') {
-                *(l+len-1) = '\0';
+        if (l != NULL) {
+                len = strlen(l);
+                if (*(l + len - 1) == '\n') {
```

Comment:
This corrupts memory if the string is empty and the character before it happens to be a `\n`.

Should this function strip all trailing newlines or just one?